src/ipc-dbus/DBusIPCAPI_v1.cpp

```diff
@@ -321,6 +329,69 @@ DBusIPCAPI_v1::received_beacon(NCPControlInterface* interface, const WPAN::Netwo
 	dbus_message_unref(signal);
 }
 
+static void
+ipc_append_network_time_update_dict(
+	DBusMessageIter *iter, const NetworkTimeUpdate& network_time_update)
+{
+	DBusMessageIter dict;
+
+	dbus_message_iter_open_container(
```

Comment:
If we change the parameter to be `const ValueMap &`, it should make building the DBus message simpler.
We can use `append_any_to_dbus_iter()` here and it would take care of building the DBus dictionary.

Reply:
Great suggestion, thanks. This has also greatly simplified my DBus client code. Pull request updated